clear-intermediate-datasets/python-runnables/clear_int_datasets/runnable.py

```diff
@@ -66,13 +66,15 @@ def run(self, progress_callback):
             rcp_outputs_dict = rcp["outputs"]
             # CASE: no input dataset
             if rcp_inputs_dict:
-                input_key = list(rcp_inputs_dict.keys())[0]
-                rcp_inputs_list = [x["ref"] for x in rcp_inputs_dict[input_key]["items"]]
-                input_datasets += rcp_inputs_list
-            output_key = list(rcp_outputs_dict.keys())[0]
-            rcp_outputs_list = [x["ref"] for x in rcp_outputs_dict[output_key]["items"]]
-            # Append them to the overall input list:
-            output_datasets += rcp_outputs_list
+                input_keys = list(rcp_inputs_dict.keys())
+                for input_key in input_keys:
+                    rcp_inputs_list = [x["ref"] for x in rcp_inputs_dict[input_key]["items"]]
+                    input_datasets += rcp_inputs_list
+            output_keys = list(rcp_outputs_dict.keys())
```

Comment:
the non-emptiness of rcp_outputs_dict should be also tested before this line

Reply:
Previous versions didn't test it and I think the reason is that, by design in DSS, a recipe requires output. It's impossible to have a recipe without output and so it's impossible to have an empty _rcp_ouputs_dic_